### DISCUSSION TITLE
Weighted average of two images

### DISCUSSION THREAD

**Discussion by DanielDrs** [ImageMagick/ImageMagick#2220] [Help] [answered]
Hallo,

I want to create a weigthed average of two images (similar to the -average operator). The new image should be 
something like 

    0.8 * (first image)  +  0.2 * (second image)

Best regards,  Daniel

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform?

I would do that with "-compose Mathematics -composite". For example:
```
magick in1.png in2.png -compose Mathematics -define compose:args=0,0.2,0.8,0 -composite out.png
```
Note the order of 0.2 and 0.8 is the opposite of what you might expect.

**Reply by fmw42** [reply]
You can also do that with -poly (which allows a multiplicative factor and an exponent), setting the exponent to 1 for each image. So

`magick in1.png in2.png -poly "0.8,1 0.2,1" out.png
`

**Comment by DanielDrs** [comment]
Thank you very much!!!  Both solutions worked for me and produced the same results.    

Best regards,  Daniel
